File: app/src/main/java/com/example/administrador/starwarswiki/FavoriteDao.java
package com.example.administrador.starwarswiki;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

@Dao
public interface FavoriteDao {
    @Insert(onConflict = REPLACE)
    void save(Favorite favorite);
    @Query("SELECT * FROM favorite WHERE starWarsCharacterId = :starWarsCharacterId")
    LiveData<Integer> load(int starWarsCharacterId);
    @Query("SELECT * FROM favorite")
    LiveData<List<Favorite>> getAllFavorites();
    @Query("DELETE FROM favorite WHERE starWarsCharacterId = :id")
    void deleteFavorite(int id);
}

File: app/src/main/java/com/example/desafio_android/data/db/FavoriteDao.kt
package com.example.desafio_android.data.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.example.desafio_android.data.model.People

@Dao
interface FavoriteDao {

    @Insert
    suspend fun addToFavorite(people: FavoritePeople)

    @Query("SELECT people FROM favorite_people")
    fun getFavorites(): LiveData<List<People>>

    @Query("SELECT count(*) FROM favorite_people WHERE favorite_people.nome = :namePeople")
    suspend fun checkPeople(namePeople: String): Int

    @Query("DELETE FROM favorite_people WHERE favorite_people.people = :people")
    suspend fun removeFromFavorite(people: People)
}